src/listener_message.rs

```diff
@@ -15,18 +15,11 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
-use sodiumoxide::crypto::sign::PublicKey;
 use socket_addr::SocketAddr;
 
+pub const REQUEST_MAGIC_CONSTANT: [u8; 4] = ['E' as u8, 'C' as u8, 'H' as u8, 'O' as u8];
```

Comment:
Ah no problem. On the encryption issue, I feel that we will have a message method/struct/enum (probably enum) that accepts all message types but to prepare them for sending at the moment we `serialise` them. I have a new crate https://github.com/maidsafe/secure_serialisation to allow fully secured messages (authenticated public key encrypted) by replacing `serialise` with `secure_serialise`. This is to make it simple to add the security we require and assuredness a message came from who we wanted (instead of sending secrets for instance in stun). So this is to make it simple and as a close to a drop in as possible. It does require a `precomputed` key (which is our `secret` key and their `public` key) to be held. This precomputed key is more efficient than always using both keys, but is also good to keep code tidy (as we refer to this single key as opposed to wondering about secret vs public etc. in code. So a single key per peer if you like.
So with that in mind to keep a drop in replacement simple as possible it seems to make sense to have the messages using a single message enum with an `impl` that defines `prepare_for_send` or similar which may call just plain `serialise` at the moment but can be easily replaced with `secure_serialisaiton::serialise` or better `secure_serialisation:: pre_computed_serialise` to allow the security and validation of messages. 
Hope that helps.